File: Chapter 03/Java/src/main/java/com/grokkingsoftwarearchitecture/chapter03/section_3_3_1_srp/after/Player.java
package com.grokkingsoftwarearchitecture.chapter03.section_3_3_1_srp.after;

/**
 * Responsibility 1: State and Core Actions.
 * * ARCHITECTURE NOTE: By stripping out the database and tactical logic, the Player 
 * class is now highly cohesive. It has only one reason to change: if the fundamental 
 * rules of a player (like adding a 'pass' method or 'stamina' property) change.
 */
public class Player {
    public String name;

    public Player(String name) {
        this.name = name;
    }

    /**
     * Executes a core domain action specific to the player's physical state.
     */
    public void dribbleBall() {
        System.out.println("  [Action] " + name + " is dribbling the ball down the court.");
    }
}

File: Chapter 03/Python/section_3_3_1_srp/after/player.py
"""
Responsibility 1: Player State and Core Actions.
* ARCHITECTURE NOTE: By stripping away infrastructure and AI logic, the Player 
class has become a "Pure Domain Model." It is now highly cohesive, meaning 
every line of code inside this class relates directly to what a player IS. 
* This is now an "Anemic-friendly" or "Rich" model that can be easily 
serialized, moved across network boundaries, or used in different game 
modes without dragging along a database connection.
"""
class Player:
    def __init__(self, name: str):
        self.name = name

    def dribble_ball(self):
        """
        Executes a core domain action.
        """
        print(f"  [Action] {self.name} is dribbling the ball down the court.")